xwiki-platform-core/xwiki-platform-localization/xwiki-platform-localization-rest/xwiki-platform-localization-rest-default/src/main/java/org/xwiki/localization/internal/DefaultTranslationsResource.java

```diff
@@ -0,0 +1,125 @@
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.LocaleUtils;
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.localization.LocalizationContext;
+import org.xwiki.localization.LocalizationManager;
+import org.xwiki.localization.Translation;
+import org.xwiki.localization.rest.TranslationsResource;
+import org.xwiki.localization.rest.model.jaxb.ObjectFactory;
+import org.xwiki.localization.rest.model.jaxb.Translations;
+import org.xwiki.model.ModelContext;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.WikiReference;
+import org.xwiki.rest.XWikiRestComponent;
+
+/**
+ * Default implementation of {@link TranslationsResource}.
+ *
+ * @version $Id$
+ * @since 13.3RC1
+ */
+@Component
+@Named("org.xwiki.localization.internal.DefaultTranslationsResource")
+@Singleton
+public class DefaultTranslationsResource implements TranslationsResource, XWikiRestComponent
+{
+    @Inject
+    private LocalizationContext localizationContext;
+
+    @Inject
+    private LocalizationManager localizationManager;
+
+    @Inject
+    private ModelContext modelContext;
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public Translations getTranslations(String wikiId, String localeString, String prefix, List<String> keys)
+    {
+        // Save the current entity reference. It will be restored on the method is finished.
+        EntityReference oldEntityReference = this.modelContext.getCurrentEntityReference();
+        try {
+            // Set the requested wiki.
+            this.modelContext.setCurrentEntityReference(new WikiReference(wikiId));
+
+            // Resolve the locale.
+            Locale locale;
+            if (localeString != null) {
+                locale = LocaleUtils.toLocale(localeString);
+            } else {
+                locale = this.localizationContext.getCurrentLocale();
+            }
+
+            // Computes the prefix.
+            String cleanPrefix = Objects.toString(prefix, "");
+
+            ObjectFactory objectFactory = new ObjectFactory();
+            Translations result = objectFactory.createTranslations();
+
+            // Resolve the raw translation of the requested keys.
+            for (String key : keys) {
+                // When no key parameter is passed to the request, the list of keys is initialized with a single null 
+                // value. See https://github.com/restlet/restlet-framework-java/issues/922
+                if (key != null) {
+                    String fullKey = cleanPrefix + key;
+                    Translation translation = this.localizationManager.getTranslation(fullKey, locale);
+                    if (translation != null) {
+                        String rawSource = (String) translation.getRawSource();
+                        result.getTranslation().add(createTranslation(objectFactory, fullKey, rawSource));
```

Comment:
This is not right. It should have generated ``getTranslations()`` (plural). So it looks like the schema is still not right.